### PR TITLE
Introduce Roslyn CFG wrapper

### PR DESCRIPTION
**Pull request by pavel-mikula-sonarsource**
This is a draft, for now, more commits with more wrappers will arrive.

### CODE REVIEW

File: analyzers/src/SonarAnalyzer.CFG/Roslyn/BasicBlock.cs [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+{
+    public class BasicBlock
+    {
+        private static readonly PropertyInfo OperationsProperty;
+
+        private readonly Lazy<ImmutableArray<IOperation>> operations;
+
+        public ImmutableArray<IOperation> Operations => operations.Value;
+
+        static BasicBlock()
```
Comment (dmitry-golovinov-sonarsource) [inline analyzers/src/SonarAnalyzer.CFG/Roslyn/BasicBlock.cs:37]: Do we really need it? Can't it be replaced with  _private static readonly PropertyInfo OperationsProperty = RoslynHelper.FlowAnalysisType("BasicBlock")?.GetProperty(nameof(Operations));_

**Review by dmitry-golovinov-sonarsource — APPROVED** [review]
LGTM with respect of fixing UTs later. Just got one minor comment.

File: analyzers/src/SonarAnalyzer.CFG/Roslyn/BasicBlock.cs (hunk 1 above)
Comment (pavel-mikula-sonarsource) [reply analyzers/src/SonarAnalyzer.CFG/Roslyn/BasicBlock.cs:37]: There will be 9 more properties coming in the next PR. For now, this class is just a dummy place holder to make UTs work.

Comment (dmitry-golovinov-sonarsource) [reply analyzers/src/SonarAnalyzer.CFG/Roslyn/BasicBlock.cs:37]: Got it

**Pull request closed without merging**